src/ui/common/src/components/pages/workflow/id/hook.ts

```diff
@@ -187,3 +188,44 @@ export function useWorkflowNodesResults(
     ),
   };
 }
+
+export function useLatestDagResultOrDag(apiKey: string, workflowId: string) {
+  const {
+    data: dagResults,
+    error: dagResultsError,
+    isLoading: dagResultsLoading,
+  } = useDagResultsGetQuery({
+    apiKey: apiKey,
+    workflowId: workflowId,
+  });
+
+  const hasRuns =
+    !dagResultsLoading && !dagResultsError && dagResults.length > 0;
+
+  const {
+    data: dags,
+    error: dagsError,
+    isLoading: dagsLoading,
+  } = useDagsGetQuery(
+    {
+      apiKey: apiKey,
+      workflowId: workflowId,
+    },
+    {
+      skip: hasRuns,
+    }
+  );
+
+  let latestDagResult;
+  let dag;
+  if (hasRuns) {
+    latestDagResult = getLatestDagResult(dagResults);
```

Comment:
if `getLatestDagResult`handles situations where `dagResults` is `undefined` or zero-length, you don't have to use  a `let` here and it gives you cleaner code. Also, whether `dag` is available depends on `dagResult`, which may makes the caller confusing (since if `dagResult` is available, caller have to look for `dag` using an additional `useDagGetQuery`. So I'd suggest call the `useDagGetQuery` here directly. Putting everything together this hook should look like:
```
const latestDagResult = getLatestDagResult(dagResults ?? []) // undefined if not available
const { data: dags } = useDagsGetQuery(..., {skip: // when latestDagResult is available or is not loaded, see line 74 of this file})
const dagIdFromLatestDagResult = latestDagResult?.dag_id;
const { data: dag } = useDagGetQuery({ dag_id: dagIdFromLatestDagResult}, {skip: // when dagIdFromLatestDagResult is not available})
return { latestDagResult, dag?? (dags ?? [])[0] // fall back to dags when dag from latestDagResult is not available)}
```

Reply:
With this change I believe you can remove most `useDagGetQuery` in the caller code